pkg/topology/kademlia/kademlia.go

```diff
@@ -967,6 +981,46 @@ func closestPeer(peers *pslice.PSlice, addr swarm.Address, spf sanctionedPeerFun
 	return closest, nil
 }
 
+func closestPeerList(peers []swarm.Address, addr swarm.Address, spf sanctionedPeerFunc) (swarm.Address, error) {
```

Comment:
since similar function already exists, can we somehow have it refactored to cater for this functionality also?

Reply:
Done.